pgfinder/matching.py

```diff
@@ -284,37 +257,31 @@ def clean_up(ftrs_df: pd.DataFrame, mass_to_clean: Decimal, time_delta: float) -
 
             for z, ins_row in ins_constrained_df.iterrows():
                 # Having cells with multiple values causes headaches! Use long format, reshape and concatenate at end if needed
-                ins_mw = list(str(ins_row.theo_mwMonoisotopic).split(","))
+                ins_mw = str(ins_row.theo_mwMonoisotopic)
 
                 # Compare parent masses to adduct masses
-                for mass in intact_mw:
-                    for mass_2 in ins_mw:
-                        mass_delta = abs(
-                            Decimal(mass).quantize(Decimal("0.00001")) - Decimal(mass_2).quantize(Decimal("0.00001"))
-                        )
-                        # Consolidate intensities
-                        if mass_delta == mass_to_clean:
-                            consolidated_decay_df.sort_values("ID", inplace=True, ascending=True)
-                            insDecay_intensity = ins_row.maxIntensity
-                            parent_intensity = row.maxIntensity
-                            consolidated_intensity = insDecay_intensity + parent_intensity
-                            ID = row.ID
-                            drop_ID = ins_row.ID
-                            idx = consolidated_decay_df.loc[consolidated_decay_df["ID"] == ID].index[0]
-                            try:
-                                drop_idx = consolidated_decay_df.loc[consolidated_decay_df["ID"] == drop_ID].index[0]
-                                consolidated_decay_df.at[idx, "maxIntensity"] = consolidated_intensity
-                                consolidated_decay_df.drop(drop_idx, inplace=True)
-                            except IndexError:
-                                #     LOGGER.info(f"Already removed : {drop_idx}")
-                                pass
-
+                mass_delta = abs(
+                    Decimal(intact_mw).quantize(Decimal("0.00001")) - Decimal(ins_mw).quantize(Decimal("0.00001"))
+                )
+                # Consolidate intensities
+                if mass_delta == mass_to_clean:
```

Comment:
I'm not sure what the intention of this block of code is which makes it hard to review. Probably a candidate for breaking out into its own function in future. I trust that as the tests pass, this is OK for now.

Reply:
It's an absolutely horrendous piece of code that took a few hours and several people to figure out, but its about consolidating the intensities of ions that exist as adducts, doing things like adding the intensity of `Na+ gm|0` structures to a parent ion `gm|0` in some retention time window.

I've added some comments and cleaned things up a bit more. Before I got to this code a few weeks ago, the consolidation had a bug that currently still exists in the `master` branch — this PR should fix that too.